homestar-runtime/src/cli/init.rs

```diff
@@ -0,0 +1,317 @@
+                .write(true)
+                .create_new(true)
+                .open(&path);
+
+            // This seemingly convoluted match is to avoid the risk of a
+            // TOCTOU race condition, where another process creates the file
+            // in between this one checking for its existence and opening it.
+            //
+            // TODO: there should probably be a flag for non-interactive use
+            // that automatically overwrites the file.
+            let settings_file = match settings_file {
+                Ok(file) => file,
+                Err(err) if err.kind() == std::io::ErrorKind::AlreadyExists => {
+                    if no_input {
+                        bail!("Aborting... settings file already exists at {:?}. Pass `--force` to overwrite it", path);
+                    }
+
+                    let should_overwrite = Confirm::new(&format!(
+                        "Settings file already exists at {:?}, overwrite?",
+                        path
+                    ))
+                    .with_default(false)
+                    .prompt()
+                    .map_err(|e| miette!(e))?;
+
+                    if !should_overwrite {
+                        bail!("Aborting... not overwriting existing settings file");
+                    }
+
+                    File::options()
+                        .read(true)
+                        .write(true)
+                        .create_new(false)
+                        .open(&path)
+                        .expect("to open settings file")
+                }
+                err => err.expect("to open settings file"),
+            };
+
+            writeln!(writer, "Writing settings to {:?}", path).expect("to write");
+
+            Ok(Box::new(settings_file))
+        }
+    }
+}
+
+fn handle_key_type(
+    key_type: Option<KeyTypeArg>,
+    no_input: bool,
+    _writer: &mut Box<dyn Write>,
+) -> Result<KeyType> {
+    match key_type {
+        Some(KeyTypeArg::Ed25519) => Ok(KeyType::Ed25519),
+        Some(KeyTypeArg::Secp256k1) => Ok(KeyType::Secp256k1),
+        None => {
+            if no_input {
+                bail!("Aborting... cannot prompt for key type in non-interactive mode. Pass `--key-type <KEY_TYPE>` to set it.");
+            }
+
+            let options = vec![KeyType::Ed25519, KeyType::Secp256k1];
+
+            let key_type = Select::new("Select key type", options)
+                .prompt()
+                .map_err(|e| miette!(e))?;
+
+            Ok(key_type)
+        }
+    }
+}
+
+fn handle_key(
+    key_arg: Option<KeyArg>,
+    key_type: KeyType,
+    no_input: bool,
+    _writer: &mut Box<dyn Write>,
+) -> Result<PubkeyConfig> {
+    let config = match key_arg {
+        None => {
+            if no_input {
+                bail!("Aborting... cannot prompt for key in non-interactive mode. Pass`--key-file <KEY_FILE>` or `--key-seed [<KEY_SEED>]` to configure the key.");
+            }
+
+            let options = vec![
```

Comment:
can this be made a const array? Or does Select for the vec anyway?

Reply:
I don't believe so, unfortunately. It moves the selected option to return it by ownership, so we'd also need to implement `Copy` over the `KeyType`, but even then, I'm only seeing a constructor over a `Vec`